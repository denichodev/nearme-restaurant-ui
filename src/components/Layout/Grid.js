import React from 'react';
import { node, bool, string } from 'prop-types';
import cn from 'classnames';

const Grid = props => {
  return (
    <div
      className={cn(
        'wrap',
        props.fluid ? 'container-fluid' : 'container',
        props.className && props.className,
      )}
    >
      {props.children}
    </div>
  );
};

Grid.defaultProps = {
  fluid: false,
  className: '',
};

Grid.propTypes = {
  fluid: bool,
  className: string,
  children: node.isRequired,
};

export default Grid;
