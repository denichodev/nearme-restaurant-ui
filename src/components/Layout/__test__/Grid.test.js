import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Grid from '../Grid';

describe('<Grid />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Grid>
        <div>Children</div>
      </Grid>,
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('has fluid class', () => {
    const wrapper = shallow(
      <Grid fluid>
        <div>Children</div>
      </Grid>,
    );

    expect(wrapper.find('.container-fluid')).toHaveLength(1);
  });

  test('has extra class', () => {
    const wrapper = shallow(
      <Grid fluid className="hello-world">
        <div>Children</div>
      </Grid>,
    );

    expect(wrapper.find('.hello-world')).toHaveLength(1);
  });
});