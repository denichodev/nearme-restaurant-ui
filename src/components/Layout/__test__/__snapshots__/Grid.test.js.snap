// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> renders correctly 1`] = `
<div
  className="wrap container"
>
  <div>
    Children
  </div>
</div>
`;
